src/libMetrics/Logging.cpp

```diff
@@ -142,25 +142,36 @@ std::shared_ptr<logs::LoggerProvider> InitNoop() {
 
 }  // namespace
 
-Logging::Logging() {
-  std::string cmp(LOGGING_ZILLIQA_PROVIDER);
+void Logging::Initialize(std::string_view identity /*= {}*/,
+                         std::string provider /*= LOGGING_ZILLIQA_PROVIDER*/) {
+  static std::array<std::string_view, 4> allowedIdentities = {
+      "seedprv", "seedpub", "dsguard", "lookup"};
+  if (std::none_of(std::begin(allowedIdentities), std::end(allowedIdentities),
+                   [identity](const auto& allowedIdentity) {
+                     return identity.find(allowedIdentity) != std::string::npos;
+                   })) {
```

Comment:
Works, but a bit convoluted? Might have been better to write an aux function (or at least a comment):

```
if (!std::any_of(std::begin(allowedIdentities), std::end(allowedIdentities), 
   /* Is this allowed? */
    [identity](const auto& allowedIdentity) { return identity.find(allowedIdentity) != std::string::npos } })) { 
     .. 
   }
```
?

Reply:
As a matter of fact that's how I initially wrote it and thought that was a bit less clear :D .... at any rate - I've moved this to an aux function and reused in Metrics.cpp & Tracing.cpp as well.